pi-geofence/src/main/java/com/ibm/pi/geofence/GeofencingJSONUtils.java

```diff
@@ -0,0 +1,283 @@
+/**
+ * Copyright (c) 2015-2016 IBM Corporation. All rights reserved.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ibm.pi.geofence;
+
+import org.apache.log4j.Logger;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Utility methods to parse one or more geofences in geojson format.
+ */
+class GeofencingJSONUtils {
+    /**
+     * Logger for this class.
+     */
+    private static final Logger log = LoggingConfiguration.getLogger(PIGeofencingManager.class.getSimpleName());
+    /**
+     * Date format used to convert dates from/to UTC format such as "2015-08-24T09:00:00-05:00".
+     */
+    static final String DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+    /**
+     * Date format used to convert dates from/to UTC format such as "2015-08-24T09:00:00.123Z".
+     */
+    private static final String TMP_DESC = UUID.randomUUID().toString();
+
+    /**
+     * Parse a list of geofences.
+     * @param json json object representing the list of fences.
+     * @return a list of {@link PIGeofence} instances.
+     * @throws Exception if a parsing error occurs.
+     */
+    static GeofenceList parseGeofences(JSONObject json) throws Exception {
+        List<PersistentGeofence> result = new ArrayList<>();
+        JSONArray features = json.getJSONArray("features");
+        for (int i=0; i<features.length(); i++) {
+            JSONObject feature = features.getJSONObject(i);
+            result.add(parseGeofence(feature));
+        }
+        if (json.has("properties")) {
+            JSONObject properties = json.getJSONObject("properties");
+            List<String> deletedCodes = null;
+            if (properties.has("deleted")) {
+                JSONArray deletedCodesJson = properties.getJSONArray("deleted");
+                deletedCodes = new ArrayList<>(deletedCodesJson.length());
```

Comment:
> TIL that you can use a generic constructor without specifying type if it can be inferred by the context.

Not with Java 7: http://docs.oracle.com/javase/7/docs/technotes/guides/language/type-inference-generic-instance-creation.html